### ISSUE TITLE
Can't build version 0.6.3

### ISSUE THREAD

**Issue by Hellseher** [GreatAttractor/imppg#2]
Hi,

I'm about to pack your project for Guix but It failed to build 
imppg - tag 0.6.3
build log:
```
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /gnu/store/rn75fm7adgx3pw5j8pg3bczfqq1y17lk-gcc-7.5.0/bin/gcc
-- Check for working C compiler: /gnu/store/rn75fm7adgx3pw5j8pg3bczfqq1y17lk-gcc-7.5.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /gnu/store/rn75fm7adgx3pw5j8pg3bczfqq1y17lk-gcc-7.5.0/bin/c++
-- Check for working CXX compiler: /gnu/store/rn75fm7adgx3pw5j8pg3bczfqq1y17lk-gcc-7.5.0/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /gnu/store/krpyb0zi700dcrg9cc8932w4v0qivdg9-pkg-config-0.29.2/bin/pkg-config (found version "0.29.2")
-- Found Boost: /gnu/store/vp2s1nxsc9vvc1ix66xpbf041f8nacxr-boost-1.72.0/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found version "1.72.0")
CMake Warning (dev) at /gnu/store/qixwn3dw6glj848hrr3h8fxm6s7wzxm3-cmake-minimal-3.16.5/share/cmake-3.16/Modules/FindwxWidgets.cmake:727 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'wxWidgets_USE_DEBUG'.
Call Stack (most recent call first):
  /gnu/store/qixwn3dw6glj848hrr3h8fxm6s7wzxm3-cmake-minimal-3.16.5/share/cmake-3.16/Modules/FindwxWidgets.cmake:776 (WX_CONFIG_SELECT_QUERY_BOOL)
  CMakeLists.txt:137 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found wxWidgets: -L/gnu/store/sg4glq9lbn40bjpbh1kdih08x1iyllzw-wxwidgets-3.0.5.1/lib;-pthread;;;-L/gnu/store/jf269s6clr6r57p8v5c3c1qkyra6apq2-mesa-20.2.4/lib;-lwx_gtk3u_adv-3.0;-lwx_gtk3u_aui-3.0;-lwx_baseu_xml-3.0;-lwx_gtk3u_core-3.0;-lwx_baseu-3.0;-lwx_gtk3u_gl-3.0 (found version "3.0.5")
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Performing Test GNU_FFAST_MATH_SUPPORTED
-- Performing Test GNU_FFAST_MATH_SUPPORTED - Success
-- Found GLEW: /gnu/store/0ilpnm42gzg8yg76iyjfziksj4k0gbmi-glew-2.1.0/include (found version "2.1.0")
-- Checking for module 'cfitsio'
--   Found cfitsio, version 3.49
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/guix-build-imppg-0.6.3.drv-0/source
make[1]: Entering directory '/tmp/guix-build-imppg-0.6.3.drv-0/source'
make[2]: Entering directory '/tmp/guix-build-imppg-0.6.3.drv-0/source'
Scanning dependencies of target imppg
make[2]: Leaving directory '/tmp/guix-build-imppg-0.6.3.drv-0/source'
make[2]: Entering directory '/tmp/guix-build-imppg-0.6.3.drv-0/source'
[  2%] Building CXX object CMakeFiles/imppg.dir/src/about.cpp.o
[  5%] Building CXX object CMakeFiles/imppg.dir/src/adv_settings_wnd.cpp.o
[  7%] Building CXX object CMakeFiles/imppg.dir/src/align_disc.cpp.o
[ 10%] Building CXX object CMakeFiles/imppg.dir/src/align_params.cpp.o
[ 12%] Building CXX object CMakeFiles/imppg.dir/src/align_phasecorr.cpp.o
[ 15%] Building CXX object CMakeFiles/imppg.dir/src/align_proc.cpp.o
[ 17%] Building CXX object CMakeFiles/imppg.dir/src/align_progress.cpp.o
[ 20%] Building CXX object CMakeFiles/imppg.dir/src/appconfig.cpp.o
[ 22%] Building CXX object CMakeFiles/imppg.dir/src/backend/cpu_bmp/cpu_bmp_core.cpp.o
[ 25%] Building CXX object CMakeFiles/imppg.dir/src/backend/cpu_bmp/cpu_bmp_proc.cpp.o
[ 27%] Building CXX object CMakeFiles/imppg.dir/src/backend/cpu_bmp/w_lrdeconv.cpp.o
[ 30%] Building CXX object CMakeFiles/imppg.dir/src/backend/cpu_bmp/w_tcurve.cpp.o
[ 32%] Building CXX object CMakeFiles/imppg.dir/src/backend/cpu_bmp/w_unshmask.cpp.o
[ 35%] Building CXX object CMakeFiles/imppg.dir/src/backend/cpu_bmp/worker.cpp.o
[ 37%] Building CXX object CMakeFiles/imppg.dir/src/batch_params.cpp.o
[ 40%] Building CXX object CMakeFiles/imppg.dir/src/batch.cpp.o
[ 42%] Building CXX object CMakeFiles/imppg.dir/src/common.cpp.o
[ 45%] Building CXX object CMakeFiles/imppg.dir/src/cursors.cpp.o
[ 47%] Building CXX object CMakeFiles/imppg.dir/src/fft.cpp.o
[ 50%] Building CXX object CMakeFiles/imppg.dir/src/formats.cpp.o
[ 52%] Building CXX object CMakeFiles/imppg.dir/src/gauss.cpp.o
[ 55%] Building CXX object CMakeFiles/imppg.dir/src/image.cpp.o
/tmp/guix-build-imppg-0.6.3.drv-0/source/src/image.cpp: In static member function ‘static std::optional<std::unique_ptr<IImageBuffer> > c_FreeImageBuffer::Create(c_FreeImageHandleWrapper&&)’:
/tmp/guix-build-imppg-0.6.3.drv-0/source/src/image.cpp:195:12: error: could not convert ‘result’ from ‘std::unique_ptr<IImageBuffer>’ to ‘std::optional<std::unique_ptr<IImageBuffer> >’
     return result;
            ^~~~~~
make[2]: *** [CMakeFiles/imppg.dir/build.make:336: CMakeFiles/imppg.dir/src/image.cpp.o] Error 1
make[2]: Leaving directory '/tmp/guix-build-imppg-0.6.3.drv-0/source'
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/imppg.dir/all] Error 2
make[1]: Leaving directory '/tmp/guix-build-imppg-0.6.3.drv-0/source'
make: *** [Makefile:152: all] Error 2
command "make" failed with status 2
```

**Comment by GreatAttractor** [comment]
Hi,
If you replace this line with:
```C++
return std::move(result);
```
are there any more compile errors?
(Turns out this is only required for GCC <8, but I'm not sure if there are some other occurrences.)

**Comment by Hellseher** [comment]
Hi,

Thanks that solved it, build passed successfully!

**Comment by GreatAttractor** [comment]
Great, I've pushed a commit with the fix (`d86e4a2`).
